### DISCUSSION TITLE
Test script to pytorch

### DISCUSSION THREAD

**Discussion by Uemerson** [voxelmorph/voxelmorph#550] [Q&A]
Hello everyone!

I need some help to obtain the Dice Score for the record made with PyTorch. There is an example for TensorFlow, but I'm looking for a PyTorch example.

https://github.com/voxelmorph/voxelmorph/blob/dev/scripts/tf/test.py

I am adapting this script to work with PyTorch; however, I am stuck at the following part:

```
# apply transform
warped_seg = transform_model.predict([moving_seg, warp]).squeeze()
```

How would I do this in PyTorch?

A part of my current script:

```
for i in range(len(img_pairs)):
    # load moving image and seg
    moving_vol = vxm.py.utils.load_volfile(
        img_pairs[i][0], np_var="vol", add_batch_axis=True, add_feat_axis=add_feat_axis
    )
    moving_seg = vxm.py.utils.load_volfile(
        seg_pairs[i][0], np_var="seg", add_batch_axis=True, add_feat_axis=add_feat_axis
    )

    # load fixed image and seg
    fixed_vol = vxm.py.utils.load_volfile(
        img_pairs[i][1], np_var="vol", add_batch_axis=True, add_feat_axis=add_feat_axis
    )
    fixed_seg = vxm.py.utils.load_volfile(seg_pairs[i][1], np_var="seg")

    moving = moving_vol
    fixed = fixed_vol

    # set up tensors and permute
    input_moving = torch.from_numpy(moving).to(device).float().permute(0, 4, 1, 2, 3)
    input_fixed = torch.from_numpy(fixed).to(device).float().permute(0, 4, 1, 2, 3)

    # predict
    moved, warp = model(input_moving, input_fixed, registration=True)
    
    # I need to apply the warp transform to the moving_seg.
    
    # to compute volume overlap (dice)
        # overlap = vxm.py.utils.dice(warped_seg, fixed_seg, labels=labels)
        # dice_means.append(np.mean(overlap))
        # print('Pair %d    Reg Time: %.4f    Dice: %.4f +/- %.4f' % (i + 1, reg_time,
        #                                                            np.mean(overlap),
        #                                                            np.std(overlap)))
```

Another question is the jacobian_determinant function in the script:

https://github.com/voxelmorph/voxelmorph/blob/dev/voxelmorph/py/utils.py

Would I feed the result of the warp variable into the script above to obtain the Jacobian Determinant j <= 0% ?

Is it correct to use the Jacobian Determinant as a metric to assess the registration efficiency?
